homework/src/index.js

```diff
@@ -31,17 +31,66 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+    const root = document.getElementById("root");
+    const div = createAndAppend("div", root, { class: "container" });
+    const header = createAndAppend("p", div, {
+      text: "HYF Repositories",
+      class: "header"
+    });
+    const table = createAndAppend("table", root, { class: "table" });
+    const select = createAndAppend("select", header, { class: "select" });
+
+    const tbody = createAndAppend("tbody", table);
+    const tr = createAndAppend("tr", tbody);
+
+    createAndAppend("td", tr, tbody, { text: "Repository: ", class: "label" });
+    const repository = createAndAppend("a", tbody);
+    const description = createAndAppend("tr", tbody);
+    const forks = createAndAppend("tr", tbody, { class: "label" });
+
+    const updated = createAndAppend("tr", tbody, { class: "label" });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend("div", root, {
+          text: error.message,
+          class: "alert-error"
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
       }
+
+      repositories.forEach((element, index) => {
```

Comment:
If the server returns an error, say a 404 "Not Found" error, your code still attempts to continue after rendering an error in the HTML page. You can see this error message in the console:

```
index.js:63 Uncaught TypeError: Cannot read property 'forEach' of undefined
    at fetchJSON (index.js:63)
    at XMLHttpRequest.xhr.onload (index.js:12)
```

A possible solution could be to simply return from your `main()` function after rendering the error. In that case, you also do not need to use `else`:

```js
fetchJSON(url, (error, repositories) => {
  if (error) {
    createAndAppend("div", root, {
      text: error.message,
      class: "alert-error"
    });
    return;
  }

  repositories.sort((a, b) => a.name.localeCompare(b.name));
  repositories.forEach((element, index) => {
  //...
```